main.cpp

```diff
@@ -118,23 +108,9 @@ struct DoubleType
     DoubleType& subtract(double rhs );
     DoubleType& multiply(double rhs );
     DoubleType& divide(double rhs );
-    // 4 functions taking a FloatType as input
-    DoubleType& add(const FloatType& rhsFt );
-    DoubleType& subtract(const FloatType& rhsFt );
-    DoubleType& multiply(const FloatType& rhsFt );
-    DoubleType& divide(const FloatType& rhsFt );
-    // 4 functions taking a DoubleType as input
-    DoubleType& add(const DoubleType& rhsDt );
-    DoubleType& subtract(const DoubleType& rhsDt );
-    DoubleType& multiply(const DoubleType& rhsDt );
-    DoubleType& divide(const DoubleType& rhsDt );
-    // 4 functions taking an int as input
-    DoubleType& add(const IntType& rhsIt );
-    DoubleType& subtract(const IntType& rhsIt );
-    DoubleType& multiply(const IntType& rhsIt );
-    DoubleType& divide(const IntType& rhsIt );
+    // convert UDT to primitive type
+    operator double();
```

Comment:
It's easiest to just implement these in-class
- [ ] move conversion function implementation to inside of class.